tests/unit/amo/sagas/test_collections.js

```diff
@@ -324,140 +325,220 @@ describe(__filename, () => {
-        user: 'some-user',
+        user,
         ...params,
       }));
     };
 
-    it('sends a patch to the collection API', async () => {
-      const params = {
-        collectionSlug: 'a-collection',
-        description: { 'en-US': 'New collection description' },
-        name: { 'en-US': 'New collection name' },
-        slug: 'new-slug',
-        user: 543,
-      };
-      const state = sagaTester.getState();
+    describe('common logic', () => {
+      // The common logic can be tested either by dispatching updateCollection
+      // or createCollection, so we just use _updateCollection.
+      it('clears the error handler', async () => {
+        _updateCollection();
 
-      mockApi
-        .expects('updateCollection')
-        .withArgs({
-          api: state.api,
-          collectionSlug: params.collectionSlug,
-          defaultLocale: undefined,
-          description: params.description,
-          name: params.name,
-          slug: params.slug,
-          user: params.user,
-        })
-        .once()
-        .returns(Promise.resolve());
+        const expectedAction = errorHandler.createClearingAction();
+
+        const action = await sagaTester.waitFor(expectedAction.type);
+        expect(action).toEqual(expectedAction);
+      });
 
-      _updateCollection(params);
+      it('handles errors', async () => {
+        const error = new Error('some API error maybe');
 
-      const expectedAction = pushLocation('/example/collection/url/');
-      await sagaTester.waitFor(expectedAction.type);
+        mockApi
+          .expects('updateCollection')
+          .returns(Promise.reject(error));
 
-      mockApi.verify();
+        _updateCollection();
+
+        const expectedAction = errorHandler.createErrorAction(error);
+        const action = await sagaTester.waitFor(expectedAction.type);
+        expect(action).toEqual(expectedAction);
+      });
     });
 
-    it('deletes collection object after successful update', async () => {
-      mockApi.expects('updateCollection').returns(Promise.resolve());
+    describe('update logic', () => {
+      it('sends a request to the collection API', async () => {
+        const params = {
+          collectionSlug: 'a-collection',
+          description: { 'en-US': 'New collection description' },
+          name: { 'en-US': 'New collection name' },
+          slug: 'new-slug',
+          user,
+        };
+        const state = sagaTester.getState();
+
+        mockApi
+          .expects('updateCollection')
+          .withArgs({
+            api: state.api,
+            collectionSlug: params.collectionSlug,
+            defaultLocale: undefined,
+            description: params.description,
+            name: params.name,
+            slug: params.slug,
+            user,
+          })
+          .once()
+          .returns(Promise.resolve());
+
+        _updateCollection(params);
+
+        // TODO: Instead of this we should probably update the collectionUpdates state,
```

Comment:
Can we get an issue filed for this if it's still the case and a code quality thing we should fix in the future?

Reply:
I added issue #4717 to track this, and added a link to that issue into the TODO comment.